apps/abstraxion-dashboard/components/AbstraxionWallets/index.tsx

```diff
@@ -122,7 +84,67 @@ export const AbstraxionWallets = () => {
     } finally {
       setIsGeneratingNewWallet(false);
     }
-  };
+  }, [
+    apiUrl,
+    session_jwt,
+    session_token,
+    setIsGeneratingNewWallet,
+    startPolling,
+    setFetchingNewWallets,
+    setAbstraxionError,
+  ]);
+
+  useEffect(() => {
+    async function onStartup() {
+      if (!loading && data && !previousData && !isGeneratingNewWallet) {
+        if (
+          data?.smartAccounts?.nodes.length === 0 &&
+          connectionType === "stytch"
+        ) {
+          await handleJwtAALoginOrCreate();
+        }
+      }
+    }
+
+    onStartup();
+  }, [
+    loading,
+    data,
+    connectionType,
+    handleJwtAALoginOrCreate,
+    isGeneratingNewWallet,
+    previousData,
```

Comment:
Do we want to call this function when any of these values change?

Reply:
Adding to this, why did you break this apart? I'm curious.

Reply:
I'm not sure what you mean by break apart